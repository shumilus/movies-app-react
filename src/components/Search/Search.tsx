import React, { useEffect, useState } from 'react';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  inputContainer: {
    position: 'relative',
    width: '100%',
    marginRight: '15px',
    borderRadius: '4px',
  },
  input: {
    position: 'relative',
    zIndex: 2,
    background: 'transparent',
    width: '100%',
    fontSize: '20px',
    mixBlendMode: 'normal',
    borderRadius: '4px',
    fontWeight: '400',
    lineHeight: '24px',
    padding: '17px 20px',
    boxSizing: 'border-box',
  },
  inputBackground: {
    position: 'absolute',
    background: 'rgba(50, 50, 50, 0.8)',
    zIndex: 1,
    width: '100%',
    height: '100%',
    opacity: '0.7',
    borderRadius: '4px',
  },
  button: {
    padding: '17px 73px',
  },
});

interface SearchProps {
  search: string;
  onSearchClick: (value: string) => void;
}


export default function Search({ search, onSearchClick }: SearchProps) {
  const [value, setValue] = useState<string>(search);
  const classes = useStyles();

  const onSearchChanged = (event: any) => {
    setValue(event.target.value);
  };

  useEffect(() => {
    setValue(search);
  }, [search]);

  return (
    <div className='d-flex'>
      <div className={classes.inputContainer}>
        <div className={classes.inputBackground}></div>
        <input type='text'
               value={value || ''}
               className={`${classes.input} light-color search-input`}
               placeholder='What do you want to watch?'
               onChange={onSearchChanged}/>
      </div>

      <button
        className={`${classes.button} primary-button`}
        onClick={() => onSearchClick(value)}>Search</button>
    </div>
  );
}
