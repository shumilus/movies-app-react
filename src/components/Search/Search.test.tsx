import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Search from './Search';
import userEvent from '@testing-library/user-event';

describe('Search', () => {
  it('should match Search snapshot', () => {
    const handleClick = jest.fn();

    const { asFragment } = render(
      <Router>
        <Search search='search' onSearchClick={handleClick}/>
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onSearchClick on button click', () => {
    const handleClick = jest.fn();

    render(<Search search='Action' onSearchClick={handleClick} />);
    userEvent.click(screen.getByRole('button'));

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
