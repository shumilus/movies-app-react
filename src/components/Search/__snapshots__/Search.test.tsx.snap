// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match Search snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex"
  >
    <div
      class="inputContainer-0-2-1"
    >
      <div
        class="inputBackground-0-2-3"
      />
      <input
        class="input-0-2-2 light-color"
        placeholder="What do you want to watch?"
        type="text"
        value="search"
      />
    </div>
    <button
      class="button-0-2-4 primary-button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
